node/src/main.rs

```diff
@@ -128,13 +130,27 @@ fn async_main() -> impl Future<Item = (), Error = ()> + Send + 'static {
 
     info!(logger, "Starting up");
 
-    // Create system components
+    // Create and test IPFS client
+    info!(logger, "Connecting to IPFS node...");
     let resolver = Arc::new(
         IpfsClient::new(
             &format!("{}", ipfs_socket_addr.ip()),
             ipfs_socket_addr.port(),
         ).expect("Failed to start IPFS client"),
     );
+    let ipfs_test = resolver.cat("/ipfs/QmS4ustL54uo8FzR9455qaxZwuMiUhyvMcX9Ba8nUH4uVv/readme");
+    if let Err(e) = ipfs_test.concat2().wait() {
+        error!(logger, "Failed to connect to IPFS: {}", e);
+        error!(
+            logger,
+            "Is there an IPFS node running at '{}'?", ipfs_socket_addr
+        );
+        thread::sleep(Duration::from_millis(50)); // small delay to let logger finish writing
```

Comment:
This is an arbitrary timeout that doesn't guarantee the message gets logged. Is there a way to flush the logger or to panic through the logger?

Reply:
Yeah, I know :(
I looked long and hard for a way to flush with `slog` but couldn't find anything. Without this, both the panic message and the logger messages are printed, but they overlap sometimes, so it's really just an issue of polish.

This is a general problem that applies to all panics in this function. Maybe I could wrap the call to async_main in [`catch_unwind`](https://doc.rust-lang.org/std/panic/fn.catch_unwind.html) and log any panic messages manually through `slog`?

Reply:
You can try dropping `logger` before panicking or creating and dropping an [AsyncGuard](https://docs.rs/slog-async/2.3.0/slog_async/struct.AsyncBuilder.html#method.build_with_guard), it's supposed to guarantee flushing.

Reply:
Panicking here in this function is not the best behavior anyways because it will not actually crashed the node, but just leave it hanged in some weird state.